### PR TITLE
GH-9623 GetFilePreview method added to plugin API

### PR DESCRIPTION
**Pull request by s4kh**
#### Summary
 GetFilePreview method added to plugin API

#### Ticket Link
#9623 

#### Checklist
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

**Comment by crspeller** [comment]
@S4KH You have some build failures.

**Comment by S4KH** [comment]
Following errors happened: 
```
app/plugin_api.go:327:2: not enough arguments to return
app/plugin_api.go:327:16: api.app.GetFilePreview undefined (type *App has no field or method 
GetFilePreview)
```

Should not there be an implementation inside `file.go` inside `app` package?

**Comment by crspeller** [comment]
@S4KH There is not. If you check there is no GetFilePreview. You can check `api4/file.go` and to see how the file preview is returned in the API and create one based on that.

**Comment by S4KH** [comment]
@crspeller Thanks for the quick response. Does that mean I have to create a function inside `app/file.go` using the API function inside `api4/file.go`?

**Comment by S4KH** [comment]
Can you provide an example code if it is possible?

File: app/plugin_api.go [hunk 1]
```diff
@@ -361,6 +362,29 @@ func (api *PluginAPI) GetFileInfo(fileId string) (*model.FileInfo, *model.AppErr
...
+	info, err := api.app.GetFileInfo(fileId)
+	if err != nil {
+		return nil, err
+	}
+
+	if info.PreviewPath == "" {
+		return nil, model.NewAppError("GetFilePreview", "plugin_api.get_file_preview.no_preview.app_error", nil, "file_id="+info.Id, http.StatusBadRequest)
+	}
+
+	fileReader, err := api.app.FileReader(info.PreviewPath)
```
Comment (lieut-data) [inline app/plugin_api.go]: If we're just reading it into a buffer to return, can we just use `api.app.ReadFile(path)` like on line 389?

@crspeller, should we be looking into more complex implementations here that involve passing a reader to the plugin instead, to help with larger files?

File: plugin/api.go [hunk 2]
```diff
@@ -214,6 +214,9 @@ type API interface {
 	// GetFileInfo gets a File Info for a specific fileId
 	GetFileInfo(fileId string) (*model.FileInfo, *model.AppError)
 
+	// GetFilePreview gets the bytes for a file by id.
```
Comment (lieut-data) [inline plugin/api.go]: Can you add a comment documenting the minimum version here?
```suggestion
	// GetFilePreview gets the bytes for a file by id.
	//
	// Minimum server version: 5.6
```

**Pull request closed without merging**